cdk/hls_constructs/aws_batch_infra.py

```diff
@@ -56,10 +61,13 @@ def __init__(
             ec2.MultipartBody.from_user_data(command_user_data)
         )
 
-        # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/create-launch-template.html#use-an-ssm-parameter-instead-of-an-ami-id
-        ec2_machine_image = ec2.MachineImage.resolve_ssm_parameter_at_launch(
-            "/mcp/amis/aml2-ecs",
-        )
+        if "resolve:ssm:" in ami_id:
+            # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/create-launch-template.html#use-an-ssm-parameter-instead-of-an-ami-id
+            ec2_machine_image = ec2.MachineImage.resolve_ssm_parameter_at_launch(
+                ami_id.lstrip("resolve:ssm:")
```

Comment:
You want `removeprefix`, as `lstrip` simply removes all leading characters that are one of the given characters. For example, `"::eelmorsssv".lstrip("resolve:ssm:")` evaluates to an empty string.

This might be a moot point in this case, since the suffix likely starts with a `/`, but doesn't have to.

Reply:
Yep, that one's bitten me too. 😬